src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -170,7 +207,20 @@ public void saveCondition() {
 	@Then("the system adds New Condition in Conditions table")
 	public void systemAddsCondition() {
 		assertNotNull(conditionsPage.getConditionsList());
+	}
+
+	@And("a user clicks on the delete button from dashboard")
+	public void deleteCondition() {
+		conditionsPage.deleteFirstActive();
+        conditionsPage.confirmDeleteCondition();
+	    dashboardPage = conditionsPage.clickReturn();
+	}
+
+	@Then("the system deletes a condition from the conditions table")
+	public void confirmDeleteCondition() {
+		conditionsPage.confirmDeleteCondition();
 		dashboardPage = conditionsPage.clickReturn();
+
```

Comment:
isn't the condition already deleted in the previous step? if yes, isn't the method here supposed to assert that the conditions table is not null since there remains one condition?

Reply:
> isn't the condition already deleted in the previous step? if yes, isn't the method here supposed to assert that the conditions table is not null since there remains one condition?

Hey @kdaud  are you saying I should do something like this
```
public void deleteCondition() {
	conditionsPage.deleteFirstActive();
        conditionsPage.confirmDeleteCondition();
        assertNotNull(conditionsPage.getConditionsList());
	dashboardPage = conditionsPage.clickReturn();
	}
```

Reply:
yes

Then instead confirmDeleteCondition() we can have `systemDeletesCondition()` something like

```js
public void systemDeletesCondition() {
      // assert that the conditions table isn't null since one condition is present in the table
}